akka-bbb-apps/src/main/scala/org/bigbluebutton/core/db/DatabaseConnection.scala

```diff
@@ -18,4 +27,25 @@ object DatabaseConnection {
         logger.error(s"Failed to initialize database: ${ex.getMessage}")
     }
   }
+
+  def enqueue(action: DBIO[_]): Unit = {
+    queue.add(action)
+  }
+
+  private def processBatch(): Unit = {
+    if (!queue.isEmpty) {
+      val batch = queue.iterator().asScala.toSeq
+      queue.clear() // Clear the queue
```

Comment:
Getting in one line and clearing in the other can lead to data loss.

Please try to find some solution that transfers the items to another list, avoiding the call to the "clear" method.

Maybe calling [remove](https://github.com/scala-js/scala-js/blob/main/javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala#L86) of the first item in a loop?